add Non Token Gated - Manual Approval Required flow

J-Son89 opened the pull request:
fixes: https://github.com/status-im/status-mobile/issues/14459

This implements the copy for the flow of Non Token Gated - Manual Approval Required

To **test** this:

Go to the community screen of a community which requires that an admin approves the request.

See that the copy is adjusted when manual approval is required.
It should match the flow of the designs here:
https://www.figma.com/file/h9wo4GipgZURbqqr1vShFN/Communities-for-Mobile?node-id=7035%3A462899&t=jzuZepNc60qisfSe-0

J-Son89 committed:
chore: add text for closed community overview page

J-Son89 committed:
adjust subs

J-Son89 commented:
<img width="391" alt="Screenshot 2022-12-09 at 13 34 16" src="https://user-images.githubusercontent.com/22799766/206716411-83f6415e-b637-4129-8d83-844b08731534.png">

<img width="389" alt="Screenshot 2022-12-09 at 13 34 10" src="https://user-images.githubusercontent.com/22799766/206716414-5f50660d-ab46-4350-9086-de5acb120008.png">

<img width="374" alt="Screenshot 2022-12-09 at 13 34 01" src="https://user-images.githubusercontent.com/22799766/206716417-0f678938-1c3f-4925-bf07-274a812b6fb2.png">

J-Son89 commented on src/quo2/components/community/icon.cljs:
```diff
@@ -0,0 +1,11 @@
+(ns quo2.components.community.icon
```
I moved the community icon to quo2 @flexsurfer 

Removed a lot of the excess code in it, this doesn't handle the "Status" default community icon, but imo we should move away from this community and start using real data as we are adding to many branches to the code just for this community

J-Son89 commented on src/status_im2/contexts/communities/discover/view.cljs:
```diff
@@ -116,10 +115,9 @@
     (fn []
-      (let [communities (rf/sub [:communities/communities])
-            communities-count (count communities)
-        ; TODO move sorting to subscription
```
actually, do we really want sorting in the subs? This seems like unnecessary computation we probably don't want for composing subs.. Right @ilmotta , @cammellos , @flexsurfer ? 🤔

flexsurfer replied:
actually its my comment, views should be dumb, all computation should be in subscriptions, so there should be `:communities/sorted-discover-communities` subscription

ilmotta replied:
I prefer to think that sorting is not the view responsibility. The view just cares about getting the data and transforming it to Hiccup.

If we move the sorting logic out of the subscription, we actually lose the caching facilities because now sort will be computed on every render (and they happen a lot).

J-Son89 commented on src/status_im2/contexts/communities/requests/actions/view.cljs:
```diff
@@ -79,16 +85,17 @@
-  (let [agreed-to-rules? (reagent/atom false)]
+(defn request-to-join [{:keys [permissions name id]}]
+  (let [agreed-to-rules? (reagent/atom false)
+        is-open? (not= 3 (:access permissions))]
```
backend permissions for closed/open community. It would probably be nicer if they used a semantic naming for this instead. Can we provide feedback on that? @cammellos, @ilmotta 
wdyt?

ilmotta replied:
Great question @J-Son89. As usual, it depends, but I stand firmly in the ground that we should almost always create a well defined model of the data coming INTO the application. If the backend sends a `3`, `:three` or `1+1+1` it doesn't matter, our data model should make sure the so called `3` is transformed into something with semantic meaning as you said.

I made a comment above related to this magic number 3, I think there's another place where this number appears.

At the very least, we could extract the [magic number](https://en.wikipedia.org/wiki/Magic_number_(programming)) to a constant with a proper name.

We also have the `src/data_store/` directory, where various namespaces exist to implement some kind of adapter pattern or anti-corruption layer.

cammellos replied:
Yes, this should be definitely `const` somewhere (we have many in `constants` namespace), we use enums in status-go because there is more idiomatic, but here is best to always use a reference/variable rather than the actual value

J-Son89 replied:
yep, it was already there. Found it, thanks! :)

J-Son89 commented on src/status_im2/subs/communities.cljs:
```diff
@@ -77,13 +77,19 @@
    (filterv
     (fn [{:keys [name featured id]}]
-      (and (or featured (= name "Status"))                  ;; TO DO: remove once featured communities exist
+      (and (or featured (= name "Status")) ;; TO DO: remove once featured communities exist
```
will revert this change, but really I would prefer to nuke this Status community from subs 👿

flexsurfer replied:
i think thats fine to nuke why not ?

J-Son89 replied:
ok, nuking! =D

status-im-auto commented:
## 0% of end-end tests have passed
```
Total executed tests: 1
Failed tests: 1
Passed tests: 0
Not executed tests: 9
```
```
IDs of not executed tests: 702730,702731,702745,702733,702774,702775,702776,702807,702808 
```
```
IDs of failed tests: 702732 
```
<h3>Not executed tests (9)</h3><details><summary>Click to expand</summary><br/><li><a href="https://ci.status.im/job/status-mobile/job/e2e/job/status-app-prs-rerun/parambuild/?BRANCH_NAME=develop&APK_NAME=14513.apk&PR_ID=14513&TR_CASE_IDS=702730,702731,702745,702733,702774,702775,702776,702807,702808">Rerun not executed tests</a></li></details><h3>Failed tests (1)</h3><details><summary>Click to expand</summary><br/><li><a href="https://ci.status.im/job/status-mobile/job/e2e/job/status-app-prs-rerun/parambuild/?BRANCH_NAME=develop&APK_NAME=14513.apk&PR_ID=14513&TR_CASE_IDS=702732">Rerun failed tests</a></li><br/><h4>Class TestGroupChatMediumMultipleDeviceNewUI:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3065135">test_group_chat_pin_messages</a>, id: 702732 </b></td></tr><tr><td><p><blockquote><div>Device 2: Find `Button` by `accessibility id`: `open-activity-center-button` </div><div>Device 2: Tap on found: Button </div></blockquote></p>

```
Test setup failed: critical/chats/test_group_chat.py:227: in prepare_devices
    self.group_chat_2 = self.home_2.get_chat(self.group_chat_name).click()
../views/home_view.py:265: in get_chat
    chat_in_ac.wait_for_element(20)
../views/base_element.py:116: in wait_for_element
    raise TimeoutException(
 Device `2`: `ActivityCenterChatElement` by` xpath`: `//*[@content-desc='chat-name-or-sender-text'][starts-with(@text,'Group Chat')]/../..` is not found on the screen after wait_for_element
```

<br/><br/><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/b02a3255b15940c9b2b2da90c691ad15?auth=6a8c0dc3a535e0d86810c77a96535dc2">Steps, video, logs</a></li><li><a href="https://assets.eu-central-1.saucelabs.com/jobs/b02a3255b15940c9b2b2da90c691ad15/0021screenshot.png?auth=6a8c0dc3a535e0d86810c77a96535dc2">Failure screenshot</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/bf536a5da4ed4176b2e9657638120a10?auth=bd22fc209be2f75b540c414906c692f5">Steps, video, logs</a></li><li><a href="https://assets.eu-central-1.saucelabs.com/jobs/bf536a5da4ed4176b2e9657638120a10/0014screenshot.png?auth=bd22fc209be2f75b540c414906c692f5">Failure screenshot</a></li></ul></p></ul></p></td></tr></tbody></table></details>

flexsurfer commented on src/status_im2/contexts/communities/overview/view.cljs:
```diff
@@ -6,8 +6,6 @@
    [utils.re-frame :as rf]
    [quo2.foundations.colors :as colors]
    [status-im2.contexts.communities.overview.style :as style]
```
please keep ;; TODO move to status-im2 when reimplemented, this comment for all requirement under it

J-Son89 replied:
I decided to remove the code that needed these imports as it was just for mock data. So no more status_im in this file 🧹

flexsurfer commented on src/status_im2/contexts/communities/overview/style.cljs:
```diff
@@ -14,4 +14,19 @@
\ No newline at end of file
+                                  :flex 1}})
+
+(defn scroll-view-container [border-radius] {:position :absolute
```
could you please move map to the new line?

ilmotta commented on src/quo2/components/community/icon.cljs:
```diff
@@ -0,0 +1,11 @@
+                        :border-width 0
+                        :border-color :transparent
+                        :width size
+                        :height size}}]))
```
At some point we need to at least agree about trailing newlines at the end of files and configure our editors properly. I think the arguments are strongly in favor of a trailing newline, no exceptions. Even Github is hinting at us with a red icon ;)

Suggestion for reading:
https://unix.stackexchange.com/questions/18743/whats-the-point-in-adding-a-new-line-to-the-end-of-a-file

J-Son89 replied:
Yeah  I agree that we need to agree, but I don't think this is something I (or any other dev) should have to even think about. `make lint-fix` should address any of these formatting issues and any errors we find in p.r's should be handled by this tool. In the case there's some formatting we don't like the tool should be updated.

Perhaps I am being too optimistic and this is not possible in our current tech stack?? 🤔 

@yqrashawn - the auto formatter you are working on can handle this case?

yqrashawn replied:
This kind of setting is usually set in editor. There's https://editorconfig.org/ for this, but generally user need to install an editor-config plugin for that.

VS Code https://stackoverflow.com/a/54270768/5671288
IntelliJ https://stackoverflow.com/a/16761228/5671288
seems ~Emacs and~ vim does this by default
Emacs check `require-final-newline`

ilmotta replied:
I think there are two levels of automation here.

1. The developer's text editor/IDE applies basic formatting during development. This is nearly instantaneous. Newlines at the end of files should be handled at this level.
2. The code format tool (zprint for example) checks the entire codebase and see if it's formatted appropriately. This is slow and we should rely on it only as a gate keeper for continuous integration.

Hence, relying only on `make lint` would be impractical IMO (way too slow). Even in Javascript land, users usually run Prettier or another formatter on every save of the file and only rely on `make lint` for CI. This is what zprint would bring to us that @yqrashawn is working on.

J-Son89 replied:
yeah, I agree. Imo we are just adding noise to our p.r's with formatting issues, when we could automate this process. if we can not automate this with `make lint-fix` then at the very least `make lint` should catch these issues and it is up to the dev to "manually" fix these issues. 
i.e the link check step on ci jobs will fail and this will notify the dev. Reviews can then focus on improving the quality of the code.

Is it just a case of we're not quite there yet, or is there something I am missing here?

J-Son89 replied:
you mean, Hence, relying only on `make lint-fix` .. ?

ilmotta replied:
Maybe my comment wasn't clear enough. We will always rely on `make lint`, but as the last step to check. Relying on `make lint-fix` would be suboptimal during development, unless you (or any other dev) prefer that way. I never want to use `make lint-fix` in my daily development, I want my editor to automatically format the file I'm currently editing.

`make lint-fix` is slow compared to instant updates provided by IDEs/editors and you'll have to run it either on every commit or generate separate "fix lint" commits, but of course, it's up to you. Solutions with commit hooks fall in the same category, they make my interactions with git slow and I would disable them during development.

> yeah, I agree. Imo we are just adding noise to our p.r's with formatting issues

> Is it just a case of we're not quite there yet, or is there something I am missing here?

Unfortunately getting the formatter correct and capable of handling our own preferences (Status contributor preferences) is far from trivial and @yqrashawn had to actually report various issues to the maintainer of the tool. @yqrashawn just now opened a PR with new changes to introduce zprint o/

yqrashawn replied:
It's definitely possible to check this in `make lint` and fix this in `make lint-fix`.

ilmotta commented on src/status_im2/contexts/communities/overview/view.cljs:
```diff
@@ -159,11 +157,14 @@
 
-(defn render-page-content [{:keys [name description locked joined id images
-                                   status tokens tags community-color]
+(def request-to-join-text (memoize (fn [is-open?] (if is-open? (i18n/label :t/join-open-community) (i18n/label :t/request-to-join-community)))))
```
This line is unreadable to me, could you please break it into multiple lines?

cammellos replied:
Also, not sure memoizing is worth it here, just out of curiosity, have you tested the difference in speed? 
(a quick and dirty way, which should give you an idea on whether they are at least comparable is doing something like:
```
(time (dotimes (range 0 10000) (get {:a 1} :a))) // insert inside the operation you care about
```
They both look like map lookups I suppose, so I'd expect both to run in constant time, though I could be wrong

ilmotta replied:
You are right, and memoizing what's memoized is no good at all. `i18n/label` expensive calls are already memoized, it's just an overhead to memoize `request-to-join-text`.

In essence, here we're memoizing an `if` expression checking a boolean, the rest is already memoized. And of course we shouldn't memoize boolean checks.

J-Son89 replied:
I will remove the memo 👍

ilmotta commented on src/status_im2/contexts/communities/overview/view.cljs:
```diff
@@ -159,11 +157,14 @@
                             :as   community}
                            channel-heights first-channel-height]
-  (let [thumbnail-image (get-in images [:thumbnail])]
+  (let [is-open? (not= 3 (:access permissions))
```
What is this [magic number](https://en.wikipedia.org/wiki/Magic_number_(programming)) `3`? It looks like it should be refactored into a constant.

J-Son89 replied:
resolved above

ilmotta commented on src/status_im2/contexts/communities/overview/view.cljs:
```diff
@@ -238,19 +239,24 @@
+                                       {:content (constantly [requests.actions/request-to-join community])
+                                        :content-height 300}])
+             :override-background-color community-color
+             :style
```
This part looks confusing, the other keys have the values in the same line, so could you join the `:style` line with the next one?

```clojure
 :style {:width        "100%"
         :margin-top   20
         :margin-left  :auto
         :margin-right :auto}
```

J-Son89 replied:
sure will do, I'm really waiting for the auto formatter 😢

ilmotta replied:
Haha Me too, it'll come

ilmotta commented on src/status_im2/contexts/communities/requests/actions/view.cljs:
```diff
@@ -33,6 +33,12 @@
                        :title "No offensive names and profile pictures"
                        :content "You will be asked to change your name or picture if the staff deems them inappropriate."}])
 
+(def request-to-join-text (memoize
```
I don't think we need to memoize this function. The expensive `i18n/label` calls are already memoized. It's often a mistake to memoize what's already memoized, it causes more overhead for nothing.

J-Son89 commented on src/status_im2/contexts/communities/home/view.cljs:
```diff
@@ -14,7 +14,7 @@
                        (rf/dispatch [:communities/load-category-states id])
                        (rf/dispatch [:dismiss-keyboard])
-                       (rf/dispatch [:navigate-to-nav2 :community {:community-id id}]))
+                       (rf/dispatch [:navigate-to :community-overview id]))
```
@flexsurfer, @jo-mut updated the navigation on the communities home page so it goes directly to the new community overview page

flexsurfer replied:
cc @Parveshdhull

Parveshdhull replied:
Hi @J-Son89,
Please use `(re-frame/dispatch [:communities/navigate-to-community {:community-id (:id community)}])`
and change `:community` to `:community-overview` in`:communities/navigate-to-community` event.

> so it goes directly to the new community overview page

Also, is the new community overview functional? I tried PR and looks like the community screen just has mocked data and channels are not opening. Is this intentional?

J-Son89 replied:
thanks @Parveshdhull, it's not functional just yet. It is mostly using mock data. However I think it should be  it's a small adjustment to get it working.

For now, I will revert this change in this p.r and look to make this and other necessary changes in a follow up pr

J-Son89 replied:
@vkjr - just for reference :)

J-Son89 commented on src/status_im2/subs/communities_test.cljs:
```diff
@@ -109,3 +109,24 @@
     (is (= {:joined ["0x1"], :pending ["0x2"], :opened ["0x3"]}
            (rf/sub [sub-name])))))
+
+(h/deftest-sub :communities/sorted-communities
```
@ilmotta added the sorted communities subscription and this test. It's alright? :)

ilmotta replied:
Looks like a useful test @J-Son89!

J-Son89 replied:
thanks for checking. and for helping set it up. Had not looked at these tests before and it was easy to figure out how to use it from the other tests 👍

status-im-auto commented:
## 100% of end-end tests have passed
```
Total executed tests: 17
Failed tests: 0
Passed tests: 17
Not executed tests: 3
```
```
IDs of not executed tests: 702807,702808,702847 
```
<h3>Not executed tests (3)</h3><details><summary>Click to expand</summary><br/><li><a href="https://ci.status.im/job/status-mobile/job/e2e/job/status-app-prs-rerun/parambuild/?BRANCH_NAME=develop&APK_NAME=14513.apk&PR_ID=14513&TR_CASE_IDS=702807,702808,702847">Rerun not executed tests</a></li></details><h3>Passed tests (17)</h3><details><summary>Click to expand</summary><br/><br/><h4>Class TestCommunityMultipleDeviceMerged:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3067946">test_community_emoji_send_copy_paste_reply</a>, id: 702840 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/3f1559f30cb643ada82050b066bbcb09?auth=70bd80a7671fa4c46aaba1b55265d829#182">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/0efb6959282347fdbecf70262880a5de?auth=c13887b5bb5fa486d04525fec15513d8#111">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3067950">test_community_links_with_previews_github_youtube_twitter_gif_send_enable</a>, id: 702844 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/3f1559f30cb643ada82050b066bbcb09?auth=70bd80a7671fa4c46aaba1b55265d829#239">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/0efb6959282347fdbecf70262880a5de?auth=c13887b5bb5fa486d04525fec15513d8#132">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>3. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3067951">test_community_leave</a>, id: 702845 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/3f1559f30cb643ada82050b066bbcb09?auth=70bd80a7671fa4c46aaba1b55265d829#286">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/0efb6959282347fdbecf70262880a5de?auth=c13887b5bb5fa486d04525fec15513d8#191">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>4. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3067947">test_community_unread_messages_badge</a>, id: 702841 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/3f1559f30cb643ada82050b066bbcb09?auth=70bd80a7671fa4c46aaba1b55265d829#209">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/0efb6959282347fdbecf70262880a5de?auth=c13887b5bb5fa486d04525fec15513d8#124">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>5. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3067945">test_community_message_delete</a>, id: 702839 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/3f1559f30cb643ada82050b066bbcb09?auth=70bd80a7671fa4c46aaba1b55265d829#160">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/0efb6959282347fdbecf70262880a5de?auth=c13887b5bb5fa486d04525fec15513d8#82">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>6. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3067948">test_community_mark_all_messages_as_read</a>, id: 702842 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/3f1559f30cb643ada82050b066bbcb09?auth=70bd80a7671fa4c46aaba1b55265d829#266">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/0efb6959282347fdbecf70262880a5de?auth=c13887b5bb5fa486d04525fec15513d8#183">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>7. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3067944">test_community_message_send_check_timestamps_sender_username</a>, id: 702838 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/3f1559f30cb643ada82050b066bbcb09?auth=70bd80a7671fa4c46aaba1b55265d829#132">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/0efb6959282347fdbecf70262880a5de?auth=c13887b5bb5fa486d04525fec15513d8#63">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>8. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3067949">test_community_message_edit</a>, id: 702843 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/3f1559f30cb643ada82050b066bbcb09?auth=70bd80a7671fa4c46aaba1b55265d829#145">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/0efb6959282347fdbecf70262880a5de?auth=c13887b5bb5fa486d04525fec15513d8#75">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestCommunityOneDeviceMerged:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3067952">test_community_navigate_to_channel_when_relaunch</a>, id: 702846 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/7801204bebd34f9dabbe771b11ae9c59?auth=5afed7b1d6930b10baeb343305536187#62">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestOneToOneChatMultipleSharedDevicesNewUi:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3067937">test_1_1_chat_text_message_delete_push_disappear</a>, id: 702733 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/0d0f34ba71d94d6ca225cf6e6787467d?auth=c2f962332fe182e21193026e103a1943#272">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/6665607e0c6047778ae2f5f6e2c3c6c9?auth=5fcd2d574c006971f48d727587dd11b9#173">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3067938">test_1_1_chat_non_latin_messages_stack_update_profile_photo</a>, id: 702745 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/0d0f34ba71d94d6ca225cf6e6787467d?auth=c2f962332fe182e21193026e103a1943#193">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/6665607e0c6047778ae2f5f6e2c3c6c9?auth=5fcd2d574c006971f48d727587dd11b9#108">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>3. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3067934">test_1_1_chat_message_reaction</a>, id: 702730 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/0d0f34ba71d94d6ca225cf6e6787467d?auth=c2f962332fe182e21193026e103a1943#83">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/6665607e0c6047778ae2f5f6e2c3c6c9?auth=5fcd2d574c006971f48d727587dd11b9#57">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>4. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3067935">test_1_1_chat_pin_messages</a>, id: 702731 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/0d0f34ba71d94d6ca225cf6e6787467d?auth=c2f962332fe182e21193026e103a1943#103">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/6665607e0c6047778ae2f5f6e2c3c6c9?auth=5fcd2d574c006971f48d727587dd11b9#78">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestGroupChatMediumMultipleDeviceNewUI:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3067936">test_group_chat_pin_messages</a>, id: 702732 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/732c266af57140e1a74a42dfd1ccac28?auth=8d65aefe719f405bbe67c396a3f062ce#64">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/01e7956631cc4d95a7db4f7f80dcce75?auth=ca15dfd385d28391c7ec53a2e0c551dc#59">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestDeeplinkOneDeviceNewUI:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3067941">test_public_chat_open_using_deep_link</a>, id: 702776 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/9949494279d34fe2a9eb99c7ea447a93?auth=812a11a7a827db3d8202d2e5895fdccb#79">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3067939">test_deep_link_with_invalid_user_public_key_own_profile_key</a>, id: 702774 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/9949494279d34fe2a9eb99c7ea447a93?auth=812a11a7a827db3d8202d2e5895fdccb#40">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>3. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3067940">test_deep_link_open_user_profile</a>, id: 702775 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/9949494279d34fe2a9eb99c7ea447a93?auth=812a11a7a827db3d8202d2e5895fdccb#97">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table></details>

pavloburykh commented:
Hey @J-Son89! Thanx for the PR. Please, take a look at the following issues:

### ISSUE 1 Cancel/Request join buttons are cut and slightly visible on Android
Tested on Android Galaxy A52, Android v. 11

**Steps:** 

1. Invite User A (Android) to community that requires approval (community A)
2. Open Discover community screen by User A 
3. Find community A in the list and open it 
4. Tap Request to join community button
5. Pay attention if Cancel/Request community buttons are visible

**Actual result:** Cancel/Request join buttons are cut and slightly visible on Android

![photo_2022-12-15 21 14 03](https://user-images.githubusercontent.com/97245802/207947353-2d15b666-491e-4e07-8aa5-97651a52653b.jpeg)

pavloburykh commented:
### ISSUE 2 Requesting approval to join community does not result in sending join request

**Steps:**

1. Create Require approval community (community A) by User A
2. Invite User B to the community
3. By User B open Discover community screen, find community A and Request a join
4. Pay attention if User B status changed to waiting for approval request and User A received a request

**Actual result:** User B is shown active Request community button. User A has not received join community request from User B. 


https://user-images.githubusercontent.com/97245802/207948149-1a815276-1403-4028-80a7-2fdc53f0fb99.mp4

pavloburykh commented:
### QUESTION Should this PR implement logic with online/offline owner nodes?

According to [Figma](https://www.figma.com/file/h9wo4GipgZURbqqr1vShFN/Communities-for-Mobile?node-id=7035%3A462899&t=d3tkKBjSTre4zqQ7-0) user should see different info and different types of buttons depending if owner node is online or offline. 

But current seems not to follow this logic and always shows Request to join community button regardless of owner node online/offline. Also, the info below the button is different from designs. 
![photo_2022-12-15 21 31 30](https://user-images.githubusercontent.com/97245802/207950265-6e8073db-5429-4ee5-a9be-148abff9904b.jpeg)

pavloburykh commented:
### ISSUE 4 Info icon on Community rules screen does not correspond to designs 

Steps: 

1. Open Community rules screen by requesting community join
2. Pay attention at info icon on Community rules screen

**Actual result:** Info icon is displayed on Community rules screen. It is not tappable and does not provide any info to User. 

![photo_2022-12-15 21 37 05](https://user-images.githubusercontent.com/97245802/207951224-2a4f0a41-cdb5-4c12-ab93-c8396c990f3b.jpeg)

Expected result: 

According to designs there is no info icon on Community rules screen. 
https://www.figma.com/file/h9wo4GipgZURbqqr1vShFN/Communities-for-Mobile?node-id=7643%3A608085&t=d3tkKBjSTre4zqQ7-0

<img width="326" alt="Communities for Mobile – Figma 2022-12-15 21-38-04" src="https://user-images.githubusercontent.com/97245802/207951448-a73055df-de08-4239-bb22-01e4517ae704.png">

J-Son89 commented:
Hi @pavloburykh , thanks very much for looking into this pr :D 

Issue 1 - (buttons not visible on join bottom sheet) will be handled in this issue: https://github.com/status-im/status-mobile/issues/14564
The reason it needs to be handled separately is because it is dependent on the new quo2 component for bottom sheet.
Could you at least scroll to that button? i.e functionality is not blocked? -> If not I will address that at least scrolling is working on android in this p.r

Issue 2 I will address as it is part of this p.r and I forgot to include this request. Good spot :) 

Issue 3 - Captcha is still being researched and it's being tracked here: 
https://github.com/status-im/status-mobile/issues/14455

Issue 4 is beyond the scope of this pr but I have created an issue to track it being implemented correctly.

J-Son89 commented:
> ### QUESTION Should this PR implement logic with online/offline owner nodes?
> According to [Figma](https://www.figma.com/file/h9wo4GipgZURbqqr1vShFN/Communities-for-Mobile?node-id=7035%3A462899&t=d3tkKBjSTre4zqQ7-0) user should see different info and different types of buttons depending if owner node is online or offline.
> 
> But current seems not to follow this logic and always shows Request to join community button regardless of owner node online/offline. Also, the info below the button is different from designs. ![photo_2022-12-15 21 31 30](https://user-images.githubusercontent.com/97245802/207950265-6e8073db-5429-4ee5-a9be-148abff9904b.jpeg)

Yeah I was also wondering about this. I will check with design team :)

cc @vkjr as he is working on the info box -

pavloburykh commented:
> Hi @pavloburykh , thanks very much for looking into this pr :D
> 
> Issue 1 - (buttons not visible on join bottom sheet) will be handled in this issue: #14564 The reason it needs to be handled separately is because it is dependent on the new quo2 component for bottom sheet. Could you at least scroll to that button? i.e functionality is not blocked? -> If not I will address that at least scrolling is working on android in this p.r
> 
> Issue 2 I will address as it is part of this p.r and I forgot to include this request. Good spot :)
> 
> Issue 3 - Captcha is still being researched and it's being tracked here: #14455
> 
> Issue 4 is beyond the scope of this pr but I have created an issue to track it being implemented correctly. #14565

Hi @J-Son89 ! Thank you for answers. Regarding ISSUE 1 - the buttons are slightly visible and I am able to tap them, so it is not a blocker for testing. 

Ping me up when ISSUE 2 and this [Question](https://github.com/status-im/status-mobile/pull/14513#issuecomment-1353607022) will be handled and I will perform re-test. Thank you!

pavloburykh commented:
ISSUE 5 List of missing accessibility ids that are required for e2e development


1. Community card element in the list of Discover community screen 
<img width="1251" alt="Appium 2022-12-19 16-48-02" src="https://user-images.githubusercontent.com/97245802/208452665-f35491dc-1ffe-4f03-95a7-64aa7131518b.png">


2. Request join/Join Community button on community screen

<img width="1254" alt="Appium 2022-12-19 16-44-28" src="https://user-images.githubusercontent.com/97245802/208451976-8e227a90-999b-428d-9e0f-7aace8af8e27.png">

3. Checkbox on Community rules screen
4. Cancel and Request join buttons on Community rules screen

<img width="385" alt="add Non Token Gated - Manual Approval Required flow by J-Son89 · Pull Request #14513 · status-im:status-mobile 2022-12-19 16-51-01" src="https://user-images.githubusercontent.com/97245802/208453346-6c61796d-d95a-4626-b3bf-b47bf7667783.png">

J-Son89 committed:
chore add event to join requested community

J-Son89 commented on src/status_im2/contexts/communities/requests/actions/view.cljs:
```diff
@@ -10,7 +10,7 @@
             [status-im.i18n.i18n :as i18n]
             [status-im.react-native.resources :as resources]
-            [status-im.ui.components.list.views :as list]
+            [status-im.utils.datetime :as datetime]
```
going to make a follow up p.r and address the rest of the imports.. didn't want to increase the size of this p.r

J-Son89 commented on src/status_im2/contexts/communities/requests/actions/view.cljs:
```diff
@@ -141,10 +158,20 @@
+                          (and can-request-access?
+                               (zero? requested-to-join-at)
+                               (can-request-access-again? requested-to-join-at))
+                          (>evt [::communities/request-to-join id])))
```
same for events, will start moving community events to the community context

J-Son89 committed:
chore add event to join requested community

J-Son89 commented:
@pavloburykh I addressed both issues and with regards to the question about the logic for showing the button and/or the notice that admin is offline.
I made it in this pr that once a request is made it should hide the `request to join button` so the user would never see both that button and the this admin is offline notice

I also added accessibility labels for the components you suggested. let me know if there is any issues with this.

pavloburykh commented:
> @pavloburykh I addressed both issues and with regards to the question about the logic for showing the button and/or the notice that admin is offline. I made it in this pr that once a request is made it should hide the `request to join button` so the user would never see both that button and the this admin is offline notice
> 
> I also added accessibility labels for the components you suggested. let me know if there is any issues with this.

Hi @J-Son89 Thanx for the fixes. 

Currently I am still reproducing [ISSUE 2](https://github.com/status-im/status-mobile/pull/14513#issuecomment-1353591562). Regarding [QUESTION](https://github.com/status-im/status-mobile/pull/14513#issuecomment-1354604488) seems like that behaviour remained the same as it was in previous build. 


https://user-images.githubusercontent.com/97245802/209000293-a0ef639b-6103-4b43-a460-a47f6557ef85.mp4

J-Son89 committed:
comunity

J-Son89 committed:
chore: adjust for review

J-Son89 commented:
@pavloburykh I updated this p.r with the changes needed. Should be working now 👍

status-im-auto commented:
## 94% of end-end tests have passed
```
Total executed tests: 18
Failed tests: 1
Passed tests: 17
Not executed tests: 2
```
```
IDs of not executed tests: 702844,702847 
```
```
IDs of failed tests: 702808 
```
<h3>Not executed tests (2)</h3><details><summary>Click to expand</summary><br/><li><a href="https://ci.status.im/job/status-mobile/job/e2e/job/status-app-prs-rerun/parambuild/?BRANCH_NAME=develop&APK_NAME=14513.apk&PR_ID=14513&TR_CASE_IDS=702844,702847">Rerun not executed tests</a></li></details><h3>Failed tests (1)</h3><details><summary>Click to expand</summary><br/><li><a href="https://ci.status.im/job/status-mobile/job/e2e/job/status-app-prs-rerun/parambuild/?BRANCH_NAME=develop&APK_NAME=14513.apk&PR_ID=14513&TR_CASE_IDS=702808">Rerun failed tests</a></li><br/><h4>Class TestGroupChatMultipleDeviceMergedNewUI:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3068983">test_group_chat_offline_pn</a>, id: 702808 </b></td></tr><tr><td><p><blockquote><div>Device 2: Tap on found: SendMessageButton </div><div>Device 3: Sending message 'message from new member' </div></blockquote></p>

```
critical/chats/test_group_chat.py:262: in test_group_chat_offline_pn
    self.chats[2].send_message(message_2)
../views/chat_view.py:866: in send_message
    self.chat_message_input.wait_for_element(wait_chat_input_sec)
../views/base_element.py:116: in wait_for_element
    raise TimeoutException(
 Device `3`: `EditBox` by` accessibility id`: `chat-message-input` is not found on the screen after wait_for_element
```

<br/><br/><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/770bc32998cc40b7921882242fb6eed0?auth=f1a62b3df578cc13904993801f36e263#128">Steps, video, logs</a></li><li><a href="https://assets.eu-central-1.saucelabs.com/jobs/770bc32998cc40b7921882242fb6eed0/0043screenshot.png?auth=f1a62b3df578cc13904993801f36e263">Failure screenshot</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/d712e8b46051409bba7e0cfb0fbf3c06?auth=48c332637c5baeff2b223b43a4d2c57a#79">Steps, video, logs</a></li><li><a href="https://assets.eu-central-1.saucelabs.com/jobs/d712e8b46051409bba7e0cfb0fbf3c06/0024screenshot.png?auth=48c332637c5baeff2b223b43a4d2c57a">Failure screenshot</a></li></ul></p><p>Device 3:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/4185d87350554e4aa60b7ecfaa5baa6c?auth=dac482a55f68cd965dfe587596894a19#68">Steps, video, logs</a></li><li><a href="https://assets.eu-central-1.saucelabs.com/jobs/4185d87350554e4aa60b7ecfaa5baa6c/0018screenshot.png?auth=dac482a55f68cd965dfe587596894a19">Failure screenshot</a></li></ul></p></ul></p></td></tr></tbody></table></details><h3>Passed tests (17)</h3><details><summary>Click to expand</summary><br/><br/><h4>Class TestDeeplinkOneDeviceNewUI:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3068981">test_public_chat_open_using_deep_link</a>, id: 702776 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/b243626a86e94b21a27df1c8adf83b56?auth=434f6c667fa834cb27b9b7566f8b94c9#80">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3068979">test_deep_link_with_invalid_user_public_key_own_profile_key</a>, id: 702774 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/b243626a86e94b21a27df1c8adf83b56?auth=434f6c667fa834cb27b9b7566f8b94c9#41">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>3. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3068980">test_deep_link_open_user_profile</a>, id: 702775 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/b243626a86e94b21a27df1c8adf83b56?auth=434f6c667fa834cb27b9b7566f8b94c9#98">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestCommunityOneDeviceMerged:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3068992">test_community_navigate_to_channel_when_relaunch</a>, id: 702846 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/09f5858184cb4875a499d2e82886b0d2?auth=9fa06e674a5f6b52c4f0848f5793d6da#63">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestOneToOneChatMultipleSharedDevicesNewUi:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3068977">test_1_1_chat_text_message_delete_push_disappear</a>, id: 702733 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/6cba92efa3474948ace696ba470bd55b?auth=fa9bc85637c3389dd59fdabb28f46e8c#271">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/95f79f0031344cbdbe3743c20f48a7fc?auth=6605c6e746fa0609e338eb4a3437d98b#174">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3068978">test_1_1_chat_non_latin_messages_stack_update_profile_photo</a>, id: 702745 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/6cba92efa3474948ace696ba470bd55b?auth=fa9bc85637c3389dd59fdabb28f46e8c#192">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/95f79f0031344cbdbe3743c20f48a7fc?auth=6605c6e746fa0609e338eb4a3437d98b#109">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>3. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3068974">test_1_1_chat_message_reaction</a>, id: 702730 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/6cba92efa3474948ace696ba470bd55b?auth=fa9bc85637c3389dd59fdabb28f46e8c#84">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/95f79f0031344cbdbe3743c20f48a7fc?auth=6605c6e746fa0609e338eb4a3437d98b#58">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>4. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3068975">test_1_1_chat_pin_messages</a>, id: 702731 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/6cba92efa3474948ace696ba470bd55b?auth=fa9bc85637c3389dd59fdabb28f46e8c#104">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/95f79f0031344cbdbe3743c20f48a7fc?auth=6605c6e746fa0609e338eb4a3437d98b#79">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestGroupChatMultipleDeviceMergedNewUI:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3068982">test_group_chat_join_send_text_messages_push</a>, id: 702807 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/770bc32998cc40b7921882242fb6eed0?auth=f1a62b3df578cc13904993801f36e263#111">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/d712e8b46051409bba7e0cfb0fbf3c06?auth=48c332637c5baeff2b223b43a4d2c57a#63">Steps, video, logs</a></li></ul></p><p>Device 3:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/4185d87350554e4aa60b7ecfaa5baa6c?auth=dac482a55f68cd965dfe587596894a19#63">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestGroupChatMediumMultipleDeviceNewUI:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3068976">test_group_chat_pin_messages</a>, id: 702732 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/741572423e92478b9c5a0b29dbfdb25e?auth=89c7a650a3868d7153b3fe6efc64bcef#65">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/eef2a4fd46364b7896193cbecf9b1450?auth=9fae5d077bfd77296cc6fad47a217cff#60">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestCommunityMultipleDeviceMerged:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3068986">test_community_emoji_send_copy_paste_reply</a>, id: 702840 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/1769ed13b0754c91b89bac16385347bb?auth=d24417cff17b29725919fa069d4aaaae#182">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/bfecd78aeec54c13bd8a53f1a5fbb3ee?auth=43e25d6971949a46a7f8b8cc5dd73ea4#112">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3068991">test_community_leave</a>, id: 702845 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/1769ed13b0754c91b89bac16385347bb?auth=d24417cff17b29725919fa069d4aaaae#247">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/bfecd78aeec54c13bd8a53f1a5fbb3ee?auth=43e25d6971949a46a7f8b8cc5dd73ea4#141">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>3. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3068987">test_community_unread_messages_badge</a>, id: 702841 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/1769ed13b0754c91b89bac16385347bb?auth=d24417cff17b29725919fa069d4aaaae#197">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/bfecd78aeec54c13bd8a53f1a5fbb3ee?auth=43e25d6971949a46a7f8b8cc5dd73ea4#125">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>4. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3068985">test_community_message_delete</a>, id: 702839 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/1769ed13b0754c91b89bac16385347bb?auth=d24417cff17b29725919fa069d4aaaae#161">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/bfecd78aeec54c13bd8a53f1a5fbb3ee?auth=43e25d6971949a46a7f8b8cc5dd73ea4#83">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>5. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3068988">test_community_mark_all_messages_as_read</a>, id: 702842 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/1769ed13b0754c91b89bac16385347bb?auth=d24417cff17b29725919fa069d4aaaae#227">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/bfecd78aeec54c13bd8a53f1a5fbb3ee?auth=43e25d6971949a46a7f8b8cc5dd73ea4#133">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>6. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3068984">test_community_message_send_check_timestamps_sender_username</a>, id: 702838 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/1769ed13b0754c91b89bac16385347bb?auth=d24417cff17b29725919fa069d4aaaae#133">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/bfecd78aeec54c13bd8a53f1a5fbb3ee?auth=43e25d6971949a46a7f8b8cc5dd73ea4#64">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>7. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3068989">test_community_message_edit</a>, id: 702843 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/1769ed13b0754c91b89bac16385347bb?auth=d24417cff17b29725919fa069d4aaaae#146">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/bfecd78aeec54c13bd8a53f1a5fbb3ee?auth=43e25d6971949a46a7f8b8cc5dd73ea4#76">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table></details>

pavloburykh commented:
> @pavloburykh I updated this p.r with the changes needed. Should be working now 👍

Hi @J-Son89 Thank you! 

Unfortunately behaviour is still the same for me. Both [ISSUE 2](https://github.com/status-im/status-mobile/pull/14513#issuecomment-1353591562) and [QUESTION](https://github.com/status-im/status-mobile/pull/14513#issuecomment-1354604488) remained. 

Could you please show a video how it works for you? Maybe we have some miscommunication on how it supposed to work. Thanx!

flexsurfer commented on src/status_im2/contexts/communities/requests/actions/view.cljs:
```diff
@@ -81,15 +88,26 @@
     :separator                         [rn/view {:margin-top 1}]
     :render-fn                         community-rule-item}])
 
+(def request-cooldown-ms (* 24 60 60 1000))
```
this probably should be somewhere in constants or in setup

J-Son89 replied:
yeah it makes sense to have it somewhere broader in scope

flexsurfer commented on src/status_im2/contexts/communities/requests/actions/view.cljs:
```diff
@@ -81,15 +88,26 @@
+           requested-to-join-at]}]
+  (let [agreed-to-rules? (reagent/atom false)
+        is-open?         (not= 3 (:access permissions))]
+    (js/console.log (not (pos? requested-to-join-at))
```
leftover

flexsurfer commented on src/status_im2/contexts/communities/requests/actions/view.cljs:
```diff
@@ -139,12 +158,25 @@
+          :type                :grey
+          :style               {:flex         1
+                                :margin-right 12}} (i18n/label :t/cancel)]
         [button/button
```
quo/button

flexsurfer commented on src/status_im2/contexts/communities/overview/view.cljs:
```diff
@@ -56,7 +52,7 @@
                     :emoji-background-color channel-color
                     :on-enter-channel       (fn []
-                                              (utils/show-popup
+                                              (js/alert
```
why ?

J-Son89 replied:
that dependency for utils came from status_im folder and this is just mock data showing a basic case of this functionality.

pavloburykh commented:
Had a call session with @J-Son89 
We figured out that request status is not updated to Pending right after request is sent. During a call I managed to pas a flow and get Pending and Joined statuses.

But during further testing I still faced a problem that described in [ISSUE 2](https://github.com/status-im/status-mobile/pull/14513#issuecomment-1353591562). Status has not been updated for me neither after reopening community card nor after re-login. 

Will be waiting updates on that from @J-Son89 Thanx for your work and patience.

J-Son89 commented:
@pavloburykh I checked and the button not disapearring after sending a request is also happening for open communities, i.e "join Community" function. As such I think this issue was brought in from previous work. I was thinking to handle this in a follow up pr as it will be cleaner to get this work in and I have some more changes to similar parts of the codebase in other p.rs.

I can create an issue for it and get it addressed this week. WDYT?

pavloburykh commented:
> @pavloburykh I checked and the button not disapearring after sending a request is also happening for open communities, i.e "join Community" function. As such I think this issue was brought in from previous work. I was thinking to handle this in a follow up pr as it will be cleaner to get this work in and I have some more changes to similar parts of the codebase in other p.rs.
> 
> I can create an issue for it and get it addressed this week. WDYT?

Hey @J-Son89! SGTM. Let's merge this PR and fix issue with request in follow up PR. Thanx for your work!

J-Son89 commented:
> > @pavloburykh I checked and the button not disapearring after sending a request is also happening for open communities, i.e "join Community" function. As such I think this issue was brought in from previous work. I was thinking to handle this in a follow up pr as it will be cleaner to get this work in and I have some more changes to similar parts of the codebase in other p.rs.
> > I can create an issue for it and get it addressed this week. WDYT?
> 
> Hey @J-Son89! SGTM. Let's merge this PR and fix issue with request in follow up PR. Thanx for your work!

Perfect, thanks Pavlo! I'll create the issue and post it here too 👍
